matron/src/events.c

```diff
@@ -145,6 +145,9 @@ MATRON_API void event_data_free(union event_data *ev) {
     case EVENT_SERIAL_REMOVE:
         free(ev->serial_remove.handler_id);
         break;
+    case EVENT_SERIAL_EVENT:
+        free(ev->serial_event.data);
```

Comment:
Set `serial_event.data` back to `NULL` to avoid potential dangling pointer issues.